Assets/Scripts/Actors/ActorComponents/ActorPhysics.cs

```diff
@@ -179,43 +169,34 @@ public bool GroundedCheck()
 	{
 		_isOnGround = false;
 
-		// spherecast down to detect when we're on the ground
-		RaycastHit hit;
-		Physics.SphereCast( transform.position + Vector3.up,
-		                    _jumpCheckRadius,
-		                    -Vector3.up * _jumpCheckDistance,
-		                    out hit,
-		                    _jumpCheckDistance,
-		                    _jumpLayer );
-		_isOnGround = ( hit.transform &&
-		                Vector3.Dot( hit.normal, Vector3.up ) > _minJumpDot );
-
-		if ( _isOnGround &&
-		     !_jumpCheck &&
-		     !IsInState( ActorStates.Climbing ) )
+		if ( !_jumpCheck )
 		{
-			if ( !IsInState( ActorStates.Grounded ) && !IsInState( ActorStates.Rolling ) )
+			// spherecast down to detect when we're on the ground
+			RaycastHit hit;
+			Physics.SphereCast( transform.position + Vector3.up,
+			                    _jumpCheckRadius,
+			                    -Vector3.up * _jumpCheckDistance,
+			                    out hit,
+			                    _jumpCheckDistance,
+			                    _jumpLayer );
+			_isOnGround = ( hit.transform &&
+			                Vector3.Dot( hit.normal, Vector3.up ) > _minJumpDot );
```

Comment:
Using the hit.normal directly could lead to not being able to jump on some weird mesh surface if there's a teeny face that's steep. That's a good reason to start making collision meshes...